### ISSUE TITLE
Reference mismatch for resolveType with classes

### ISSUE THREAD

**Issue by backbone87** [MichalLytek/type-graphql#605]
**Describe the Bug**
When returning classes from resolveType in a union type, there is a reference mismatch with the schema. may apply to other contexts of resolveType, too.

https://github.com/MichalLytek/type-graphql/blob/395357541cc4ca286a94fcad41a99c87b06df5b9/src/schema/schema-generator.ts#L803
returns a different instance of the GraphQLObjectType than what graphql-js calculates.

This makes the following check return false:
https://github.com/graphql/graphql-js/blob/v14.6.0/src/execution/execute.js#L497

this is the lookup and check that does not equal to the return value of the "resolveType" function:
https://github.com/graphql/graphql-js/blob/v14.6.0/src/execution/execute.js#L581

**To Reproduce**
```ts
@ObjectType()
class A {
  @Field()
  foo!: string;
}

@ObjectType()
class B {
  @Field()
  bar!: string;
}

const AB = createUnionType({
  name: 'AB',
  types: () => [A, B],
  resolveType: (root) => 'foo' in root ? A : B;
  // these will be translated to GraphQLObjectType's that dont match the ones that graphql-js looks up
  // (they have the same content but different instances)
});

@Resolver()
class Sub {
  @Subscription(() => AB, { ... })
  updates() {
    return { foo: 'baz' };
  }
}
```

```gql
subscription {
  updates {
    ... on A {
      foo
    }
    ... on B {
      bar
    }
  }
}
```
**Expected Behavior**
I get a subscription message with payload: `{ updates: { foo: 'bar' } }`, but i get `{ updates: {} }`

**Environment (please complete the following information):**
- OS: node:12-alpine
- Node: 12
- Package version: 0.18.0-beta.16
- TypeScript version: 3.8.3

**Additional Context**
Returning strings from resolveType does circumvent this problem.

**Comment by MichalLytek** [comment]
I have recently fixed that:
https://github.com/MichalLytek/type-graphql/issues/583

But I forgot about the custom `resolveType` - thanks for the report 💪

**Comment by MichalLytek** [comment]
> returns a different instance of the GraphQLObjectType than what graphql-js calculates

@backbone87 Do you use multiple schemas or run `buildSchema` more than once? I need more context to make a failing test case rather than applying patch in blind and waiting for your feedback if it works now 😉

**Comment by backbone87** [comment]
i use a single schema and run buildSchema once, but i use the member types (A, B) in queries and in the union type AB, which itself is used only in subscriptions

**Comment by backbone87** [comment]
fwiw: i test with pre beta.12 and it doesnt work their either

**Comment by MichalLytek** [comment]
@backbone87 
Fixed via `7bf6682`, published as `0.18.0-beta.17` 💪
